kv/notification_rule.go

```diff
@@ -91,6 +99,34 @@ func (s *Service) createNotificationRule(ctx context.Context, tx Tx, nr influxdb
 	return s.createUserResourceMapping(ctx, tx, urm)
 }
 
+func (s *Service) createNotificationTask(ctx context.Context, tx Tx, r influxdb.NotificationRule) (*influxdb.Task, error) {
+	// TODO(desa): figure out what to do about this
+	//ep, _, _, err := s.findNotificationEndpointByID(ctx, tx, r.GetEndpointID())
+	//if err != nil {
+	//	return nil, err
+	//}
+
+	// TODO(desa): pass in non nil notification endpoint.
+	script, err := r.GenerateFlux(nil)
+	if err != nil {
+		return nil, err
+	}
+
+	tc := influxdb.TaskCreate{
```

Comment:
I think this task probably needs a version too. If we change how the task is generated in the future, the version will make it easier to migrate the existing data.

Reply:
yeah for sure. Do tasks have a version?